app/views/postcodes/_postcode_lookup.html.haml

```diff
@@ -9,7 +9,7 @@
     = hidden_field_tag('constituency_id', @constituency.graph_id) if @constituency
     = hidden_field_tag('person_id', @person.graph_id) if @person
     = label_tag 'input', t('.search_by_postcode')
-    = text_field_tag(:postcode, @postcode, value: flash[:postcode], maxlength: 8, pattern:'[0-9a-zA-Z ]{5,}', id:'input')
+    = text_field_tag(:postcode, @postcode, value: flash[:postcode], maxlength: 8, pattern:'[A-Za-z]{1,2}[0-9Rr][0-9A-Za-z] ?[0-9][ABD-HJLNP-UW-Zabd-hjlnp-uw-z]{2}', id:'input')
```

Comment:
Try using the regex of ```((([A-Za-z][0-9]{1,2})|(([A-Za-z][A-Ha-hJ-Yj-y][0-9]{1,2})|(([AZa-z][0-9][A-Za-z])|([A-Za-z][A-Ha-hJ-Yj-y][0-9]?[A-Za-z]))))\s?[0-9][A-Za-z]{2})$```. This is a slightly changed version (to allow for a space, and not for Giro Bank postcodes) of the [Bulk Data Transfer postcode regex](https://www.gov.uk/government/uploads/system/uploads/attachment_data/file/488478/Bulk_Data_Transfer_-_additional_validation_valid_from_12_November_2015.pdf).